### PR TITLE
Add help search

### PR DESCRIPTION
**Pull request by mkandalf**
Adds help search.

The search matches against literal worlds (not captures) in rules. The semantics are AND for all terms, ignoring order.

This also refactors pagination to produce a consistent number of lines for showing commands (either in a context or search view).

Screen Shot 2020-06-01 at 5 12 47 PM

### CODE REVIEW

File: misc/help.talon [hunk 1]
```diff
@@ -1,6 +1,7 @@
 help alphabet: user.help_alphabet(user.get_alphabet())
 help context$: user.help_context()
 help active$: user.help_context_enabled()
+help search <phrase>$: user.help_search(phrase)
```
Comment (knausj85) [inline misc/help.talon:4]: I'd vote to user user.text here instead of phrase.

Comment (mkandalf) [reply misc/help.talon:4]: @knausj85 Updated

Comment (mkandalf) [reply misc/help.talon:4]: I think the ideal thing is to actually use a matcher consisting of all words in your commands, but I was getting some weird parses when I tried that.

Comment (knausj85) [reply misc/help.talon:4]: Looks like the help.talon thingy didn't get updated? 

`help search <user.text>$: user.help_search(user.text)`

or am I missing something?

Comment (mkandalf) [reply misc/help.talon]: My bad, didn't add the file. Check again.

**Pull request merged**